rcl/src/rcl/node.c

```diff
@@ -284,15 +284,6 @@ rcl_node_init(
     // error message already set
     goto fail;
   }
-  // The initialization for the rosout publisher requires the node to be in initialized to a point
-  // that it can create new topic publishers
-  if (rcl_logging_rosout_enabled() && node->impl->options.enable_rosout) {
-    ret = rcl_logging_rosout_init_publisher_for_node(node);
```

Comment:
this actually requires the behavior change for all client libraries rely on rcl, since `rcl_node_init` does not internally call `rcl_logging_rosout_init_publisher_for_node` internally.

- rclcpp: https://github.com/ros2/rclcpp/pull/2174
- rclpy: https://github.com/ros2/rclpy/pull/1121

https://github.com/ros2/rclc is not repo file, but needs to be addressed accordingly before this pr is merged.

Reply:
> https://github.com/ros2/rclc is not repo file, but needs to be addressed accordingly before this pr is merged.

https://github.com/ros2/rclc/pull/351